### PR TITLE
[PKG-5834] 1.27.2

### PR DESCRIPTION
**Pull request by ViridianMelody**
streamlit 1.27.2

**Destination channel:** defaults

### Links

- [PKG-5834]
- [Upstream repository](https://github.com/streamlit/streamlit)

### Explanation of changes:

- Simple dependency updates.

[PKG-5834]: https://anaconda.atlassian.net/browse/PKG-5834?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by psteyer — APPROVED** [review]

**Review by Cansisti — APPROVED** [review]
Might be good to leave a comment on why we add watchdog upper bound (introduced in newer versions)

**Comment by Cansisti** [comment]
Also `snowflake-snowpark-python` is described as
```
"snowflake-snowpark-python>=0.9.0; python_version=='3.8'",
"snowflake-connector-python>=2.8.0; python_version=='3.8'",
```

**Comment by psteyer** [comment]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -28,24 +28,28 @@ requirements:
...
+    - validators >=0.2,<1
+    # Upper bound on watchdog from later versions
     - watchdog >=2.1.5,<5  # [not osx]
     - gitpython >=3.0.7,<4,!=3.1.19
     - pydeck >=0.8.0,<1
     - tornado >=6.0.3,<7
   run_constrained:
-    - snowflake-snowpark-python >=1.17.0  # [py<312]
-    - snowflake-connector-python >=2.8.0  # [py<312]
+    - snowflake-snowpark-python >=1.17.0  # [py==38]
```
Comment (Cansisti) [inline recipe/meta.yaml:51]: `snowflake-snowpark-python >= 0.9.0` https://github.com/streamlit/streamlit/blob/1.27.2/lib/setup.py#L84
I'm wondering about `py==38` tho, if it's not too restrictive?

**Pull request merged**